src/compression.jl

```diff
@@ -101,17 +101,38 @@ function TranscodingStreams.startproc(codec::ZstdCompressor, mode::Symbol, error
             return :error
         end
     end
-    code = reset!(codec.cstream, 0 #=unknown source size=#)
+    code = reset!(codec.cstream)
     if iserror(code)
-        error[] = ErrorException("zstd error")
-        return :error
+        error[] = ErrorException("zstd error resetting compression context")
+        :error
+    else
+        :ok
+    end
+end
+
+if isdefined(TranscodingStreams, :pledgeinsize)
```

Comment:
Why do we need this? Can we not control this via version bounds?

Reply:
Yes, if the API seems good, I can [release a new version of TranscodingStreams](https://github.com/JuliaIO/TranscodingStreams.jl/pull/240) and update the compact bound here before merging this PR